subprojects/persistent-cache/src/main/java/org/gradle/cache/ManualEvictionInMemoryCache.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache;
+
+import java.util.Collection;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Function;
+
+public class ManualEvictionInMemoryCache<K, V> implements Cache<K, V> {
+    // Use 256 as initial size to start out with enough concurrency.
```

Comment:
Why is 256 enough concurrency?

Reply:
Passing an `int` to the constructor will set the concurrency level to `1`. Is this what we want?

Reply:
It will set the concurrency level to `1` in the `this` call, though the primary constructor only uses concurrency level as a sizing hint and for nothing else, so it is ignored.

Reply:
> Why is 256 enough concurrency?

256 makes the tests pass which assume the 6 artifact transforms to run in parallel. As soon as we add more elements, the size will increase. 256 seemed like a good default value to me. WDYT?